Description: restore DT_HASH tag for libc.so.6
 Should consider upstreaming a configure flag for this, perhaps.
Author: Michael Hudson-Doyle
Bug: https://sourceware.org/bugzilla/show_bug.cgi?id=29456
Last-Update: 2022-08-22
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makerules
+++ b/Makerules
@@ -558,6 +558,9 @@
 		  -Wl,--verbose 2>/dev/null | \
 	  sed > $@T \
 	      -e '/^=========/,/^=========/!d;/^=========/d' \
+              -e 's/^.*\.gnu\.hash[        ]*:.*$$/  .note.ABI-tag : { *(.note.ABI-tag) } &/' \
+              -e '/^[      ]*\.hash[       ]*:.*$$/{h;d;}' \
+              -e '/DATA_SEGMENT_ALIGN/{H;g}' \
 	      -e 's/^.*\*(\.dynbss).*$$/& \
 		 PROVIDE(__start___libc_freeres_ptrs = .); \
 		 *(__libc_freeres_ptrs) \
--- a/Makeconfig
+++ b/Makeconfig
@@ -367,6 +367,10 @@
 LDFLAGS.so += $(relro-LDFLAGS)
 LDFLAGS-rtld += $(relro-LDFLAGS)
 
+hashstyle-LDFLAGS = -Wl,--hash-style=both
+LDFLAGS.so += $(hashstyle-LDFLAGS)
+LDFLAGS-rtld += $(hashstyle-LDFLAGS)
+
 # Linker options to enable and disable DT_RELR.
 ifeq ($(have-dt-relr),yes)
 dt-relr-ldflag = -Wl,-z,pack-relative-relocs
